LanguageFeatures/Augmentation-libraries/augmented_expression_A02_t18_lib.dart

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting setters: Within an augmenting setter `augmented` must be
+///   followed by an `=` and will directly invoke the augmented setter. If
+///   augmenting a field with a setter, this will invoke the implicit setter
+///   from the augmented field.
+///
+/// @description Checks that it is a compile-time error to declare local
+/// variable named `augmented` within an augmenting setter. Test an object
+/// pattern.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmented_expression_A02_t18.dart';
+
+augment void set topLevelSetter(String value) {
+    augmented? x;
+//  ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    var f = (augmented x) {};
```

Comment:
Right, we do want to test that it is an error to use `augmented` as the name of a type in an augmenting declaration, also in the case where it is used to specify the parameter type of a function literal. But I don't see the connection to the parameter type of an augmenting setter declaration. So perhaps `(augmented x) {}` should occur in some other test?

Reply:
There is no connection with the parameter, the description was wrong (result of the copy-paste). Now description is updated.